Q1. Write a program that prints number from 1 to 10 number with 
their square.

In [1]:
i=1
while i<=10:
    print(i,end=',')
    i+=1

1,2,3,4,5,6,7,8,9,10,

Q2. Write a program that prints a table of a number, take that 
number from the user.
Sample input:
Enter a number: 5
Output:
5 x 1 = 5
5 x 2 = 10
5 x 3 = 15
5 x 4 = 20
5 x 5 = 25
5 x 6 = 30
5 x 7 = 35
5 x 8 = 40
5 x 9 = 45
5 x 10 = 50

In [2]:
i=1
n=int(input('enter numbers'))
while i<=10:
    print(n,'*',i,'=',i*n)
    i+=1

4 * 1 = 4
4 * 2 = 8
4 * 3 = 12
4 * 4 = 16
4 * 5 = 20
4 * 6 = 24
4 * 7 = 28
4 * 8 = 32
4 * 9 = 36
4 * 10 = 40


Q3. Write a program that calculates the product of all numbers 
between 1 to 10 using a while loop.

In [1]:
i=1
x=1
while i<=10:
    x*=i
    i+=1
print(x)

    

3628800


Q4. Write a program that prints numbers divisible of 7 between 1 to 
70 

In [ ]:
i=1
while i<=70:
    if i%7==0:
        print(i,end=',')
    i+=1


7,14,21,28,35,42,49,56,63,70,

Q5. Write a program that takes input from the user and print the 
reverse counting of that number.
Sample Input: 5
Sample output: 5 4 3 2 1
Note: Write the dry run and generate output of all the below 
programs by using paper and pen in your notebook.

In [39]:
n=int(input('Enter number'))
i=n
while i<=n and i>0:
    print(i,end=',')
    i-=1

5,4,3,2,1,

Q6. i = 1
while i < 20:
print(i)
i *= 2

In [41]:
i = 1
while i < 20:
    print(i)
    i *= 2

1
2
4
8
16


 Q7.
num = 3
while num ** 2 < 100:
print(num+1)
num += 2

In [42]:
num = 3
while num ** 2 < 100:
    print(num+1)
    num += 2

4
6
8
10


Q8.
i = 2
while i > -10:
print(i+2)
i -= 2

In [43]:
i = 2
while i > -10:
    print(i+2)
    i -= 2

4
2
0
-2
-4
-6


Q9. 
a = 5
ans = 1
while a != 1:
ans *= a
print(ans)
a -= 1

In [46]:
a = 5
ans = 1
while a != 1:
    ans *= a
    print(ans)
    a -= 1

5
20
60
120


Q10.
i = -10
while i < 0:
print(i**2)
i += 2


In [47]:
i = -10
while i < 0:
    print(i**2)
    i += 2

100
64
36
16
4


assignment 2

Q1.
x=5
While (x<10):
print(x+10)
x+=2

In [12]:
x=5
while (x<10):
    print(x+10)
    x+=2

15
17
19


Q2.
x=10
y=0
While x>y:
x=x-4
y+=4
print(x)

In [11]:

x=10
y=0
while x>y:
    x=x-4
    y+=4
    print(x)

6
2


Q3.
n = 3
while n >= 0:
 if n == 0:
 print("Blast off!")
 else:
 print(n)
 n -= 1



In [8]:

n = 3
while n >= 0:
 if n == 0:
    print("Blast off!")
 else:
    print(n)
 n -= 1



3
2
1
Blast off!


Q4.
total = 0
num = 5
while num > 0:
 total += num
 num -= 1
print("Total:", total)

In [7]:

total = 0
num = 5
while num > 0:
 total += num
 num -= 1
print("Total:", total)

Total: 15


Q5.
num = 10
while num >= 0:
 if num > 5:
 print(num, "is greater than 5")
 elif num == 5:
 print(num, "is equal to 5")
 else:
 print(num, "is less than 5")
 num -= 1

In [4]:

num = 10
while num >= 0:
 if num > 5:
    print(num, "is greater than 5")
 elif num == 5:
    print(num, "is equal to 5")
 else:
     print(num, "is less than 5")
 num -= 1

10 is greater than 5
9 is greater than 5
8 is greater than 5
7 is greater than 5
6 is greater than 5
5 is equal to 5
4 is less than 5
3 is less than 5
2 is less than 5
1 is less than 5
0 is less than 5


Q6.
n = 1
while n <= 15:
 if n % 3 == 0 and n % 2 == 0:
 print(n, "is a multiple of both 3 and 2")
 n += 1

In [3]:

n = 1
while n <= 15:
 if n % 3 == 0 and n % 2 == 0:
    print(n, "is a multiple of both 3 and 2")
 n += 1

6 is a multiple of both 3 and 2
12 is a multiple of both 3 and 2


Q7.
num = 12
divisor = 1
while divisor <= num:
 if num % divisor == 0:
 print(divisor)
 divisor += 1

In [2]:

num = 12
divisor = 1
while divisor <= num:
 if num % divisor == 0:
    print(divisor)
 divisor += 1

1
2
3
4
6
12


Q8.
a = 2
b = 3
while a <= 20:
 if a % b == 0:
 print(a, "is divisible by", b)
 a += 2
 b += 1

In [1]:

a = 2
b = 3
while a <= 20:
 if a % b == 0:
    print(a, "is divisible by", b)
 a += 2
 b += 1

4 is divisible by 4


Q9. Write a program to generate squares of even numbers between 1 to 10.
Output: 4,16,36,64,100

In [18]:
i=2
x=2
while i<=10:
    x=i**2
    print(x,end=',')
    i+=2

4,16,36,64,100,

Q10. Write a program to print Multiples of 3 and 5 between 1 to 30.
Output: 15, 30


In [19]:
i=1
while i<=30:
    if i%3==0 and i%5==0:
        print(i,end=',')
    i+=1

15,30,

Q11. Write a program to calculate the sum of numbers from 1 to 10 using a while
loop.
Output: 55

In [22]:
i=1
x=0
while i<=10:
    x+=i
    i+=1
print(x,end='')

55

Q12. Write a program to calculate Sum of Odd Numbers up to 30 using while loop

In [ ]:
i=1
x=0
while i<=30:
    if i%2!=0:
       x+=i
    i+=1
print(x,end='')
        
        


225

assignment-3

Q1. Write a python program to find the Sum of the Digits of a given 
number as an input.
Input Output Explanation
123 6 1+2+3 = 6
456 15 4+5+6 = 15
789 24 7+8+9 = 24

In [13]:
n=int(input('enter the no.'))
i=0
while n>0:
    i+=n%10
    n//=10
print(i)


6


Q2. Write a python program which finds the reverse of a given 
number.
Input Output Explanation
123 321 Reverse of 123 is 321.
456 654 Reverse of 456 is 654.
789 987 Reverse of 789 is 987.


In [10]:
n=int(input('enter the no.'))
i=0
while n>0:
    r=n%10
    i=i*10+r
    n//=10
print(i)

987


Q3. Write a python program which find the Factors of a given number
as an input.
Explanation: Given an integer input, the objective is to find all the factors
ofthe number by checking divisibility.
Input Output Explanation
12 1, 2, 3, 4, 6, 12 Factors of 12 are 1, 2, 3, 4, 6, and 12.
15 1, 3, 5, 15 Factors of 15 are 1, 3, 5, and 15.
18 1, 2, 3, 6, 9, 18 Factors of 18 are 1, 2, 3, 6, 9, and 18.


In [16]:
n=int(input('enter the no.'))
i=1
while i<=n:
    if n%i==0:
        print(i,end=',')
    i+=1

1,2,3,6,9,18,

Q5. Write a python program which check Whether the given number
as an input is a Palindrome or not.
Explanation: palindrome numbers are those whose reverse is equal to the 
original number.
Input Output Explanation
121 Palindrome Reverse of 121 is 121, so it’s a palindrome.
123 Not a Palindrome Reverse of 123 is 321, so it’s not a palindrome.
1331 Palindrome Reverse of 1331 is 1331, so it’s a palindrome

In [ ]:
n=int(input('enter the no.'))
x=n
i=0
while n>0:
    r=n%10
    i=i*10+r
    n//=10
if x==i:
    print('palindrome')
else:
    print('not palindrome')

not palindrome


Q6. Write a python program which finds Factorial of a Number.
Input Output Explanation
5 120 Factorial of 5 is 5! = 5×4×3×2×1 = 120.
6 720 Factorial of 6 is 6! = 6×5×4×3×2×1 = 720.
0 1 Factorial of 0 is 1 (by definition).


In [3]:
n=int(input('enter the no.'))
r=1
while n>=1:
    r*=n
    n-=1
print(r)

720


Q7. Check Whether or Not the Number is an Armstrong Number.
Explanation: An Armstrong number is a number that equals the sum of its 
digits, each raised to a power (length of that number or count of digit in that 
number).
Input Output Explanation
153 Armstrong 1³ + 5³ + 3³ = 153.
123 Not Armstrong 1³ + 2³ + 3³ ≠ 123.
370 Armstrong 3³ + 7³ + 0³ = 370.

In [6]:
n=int(input('enter the no.'))
i=n
z=0
while n>0:
    r=n%10
    r=r**3
    z=z+r
    n//=10
if z==i:
    print('Armstrong')
else:
    print('not Armstrong')

Armstrong
